--- Modules/getpath.c.orig	2021-03-11 11:47:42.769657798 +0100
+++ Modules/getpath.c	2021-03-11 11:52:26.920159439 +0100
@@ -1293,7 +1293,7 @@
     PyStatus res;
 
     /* Path: <platlibdir> / "pythonXY.zip" */
-    wchar_t *path = joinpath2(calculate->platlibdir,
+    wchar_t *path = _PyMem_RawWcsdup(
                               L"python" Py_STRINGIFY(PY_MAJOR_VERSION) Py_STRINGIFY(PY_MINOR_VERSION)
                               L".zip");
     if (path == NULL) {
